SimDataFormats/Associations/interface/MtdSimLayerClusterToRecoClusterAssociationMap.h

```diff
@@ -39,9 +42,31 @@ class MtdSimLayerClusterToRecoClusterAssociationMap {
   const_iterator end() const { return map_.end(); }
   const_iterator cend() const { return map_.cend(); }
 
+  range equal_range(const MtdSimLayerClusterRef& key) const {
+    return std::equal_range(map_.begin(), map_.end(), value_type(key, std::vector<FTLClusterRef>()), compare);
+  }
+
   const map_type& map() const { return map_; }
 
 private:
+  static bool compare(const value_type& i, const value_type& j) {
+    auto i_hAndE = (i.first)->hits_and_energies();
+    std::vector<uint64_t> i_ids(i_hAndE.size());
```

Comment:
@martinamalberti do we really need to go through the creation of these temporary vectors, that might be expensive, instead of defining a comparison function for pairs that just compare the ```.first``` element?

Reply:
@fabiocos Thanks. Changed here:
https://github.com/martinamalberti/cmssw/blob/mm_MTDAssociationMaps/SimDataFormats/Associations/interface/MtdSimLayerClusterToRecoClusterAssociationMap.h#L53-L64